src/main/java/gyro/aws/autoscaling/AutoScalingTargetTrackingConfiguration.java

```diff
@@ -0,0 +1,167 @@
+     * The predefined metric sepcification.
+     *
+     * @subresource gyro.aws.autoscaling.AutoScalingPredefinedScalingMetricSpecification
+     */
+    @Updatable
+    public AutoScalingPredefinedScalingMetricSpecification getPredefinedScalingMetricSpecification() {
+        return predefinedScalingMetricSpecification;
+    }
+
+    public void setPredefinedScalingMetricSpecification(AutoScalingPredefinedScalingMetricSpecification predefinedScalingMetricSpecification) {
+        this.predefinedScalingMetricSpecification = predefinedScalingMetricSpecification;
+    }
+
+    /**
+     * The amount of time, in seconds, after a scale in activity completes before another can start.
+     */
+    @Updatable
+    public Integer getScaleInCooldown() {
+        return scaleInCooldown;
+    }
+
+    public void setScaleInCooldown(Integer scaleInCooldown) {
+        this.scaleInCooldown = scaleInCooldown;
+    }
+
+    /**
+     * The amount of time in seconds, after a scale out activity completes before another can start.
+     **/
+    @Updatable
+    public Integer getScaleOutCooldown() {
+        return scaleOutCooldown;
+    }
+
+    public void setScaleOutCooldown(Integer scaleOutCooldown) {
+        this.scaleOutCooldown = scaleOutCooldown;
+    }
+
+    /**
+     * The target value for the metric.
+     */
+    @Updatable
+    public Double getTargetValue() {
+        return targetValue;
+    }
+
+    public void setTargetValue(Double targetValue) {
+        this.targetValue = targetValue;
+    }
+
+    @Override
+    public void copyFrom(TargetTrackingConfiguration model) {
+        setDisableScaleIn(model.disableScaleIn());
+        setEstimatedInstanceWarmup(model.estimatedInstanceWarmup());
+        setScaleInCooldown(model.scaleInCooldown());
+        setScaleOutCooldown(model.scaleOutCooldown());
+        setTargetValue(model.targetValue());
+
+        setCustomizedScalingMetricSpecification(null);
+        if (model.customizedScalingMetricSpecification() != null) {
+            AutoScalingCustomizedScalingMetricSpecification specification = newSubresource(
+                AutoScalingCustomizedScalingMetricSpecification.class);
+            specification.copyFrom(model.customizedScalingMetricSpecification());
+            setCustomizedScalingMetricSpecification(specification);
+        }
+
+        setCustomizedScalingMetricSpecification(null);
+        if (model.predefinedScalingMetricSpecification() != null) {
+            AutoScalingPredefinedScalingMetricSpecification specification = newSubresource(
+                AutoScalingPredefinedScalingMetricSpecification.class);
+            specification.copyFrom(model.predefinedScalingMetricSpecification());
+            setPredefinedScalingMetricSpecification(specification);
+        }
+    }
+
+    @Override
+    public String primaryKey() {
+        return String.format("%s", getCustomizedScalingMetricSpecification() != null ? getCustomizedScalingMetricSpecification().getName() : "");
```

Comment:
Also make target value required.
```suggestion
        return String.valueOf(getTargetValue());
```